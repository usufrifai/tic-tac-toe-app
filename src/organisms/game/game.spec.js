import React from 'react';
import Game from './index';
import renderer from 'react-test-renderer';
import Registration from '../../molecules/registration';
import Board from '../../molecules/board'; 
import  useRegistrationHook from '../../hooks/registrationHook';

jest.mock('../../hooks/registrationHook');

describe('Game', () => {

    test('Game Component with Registration form should be rendered', () => {
        useRegistrationHook.mockReturnValue({users: {}});
        const component = renderer.create(
            
            <Game 
                winnerMessage="Game Draw"
                squares={['X', 'O', 'O', 'X', 'X', 'O', 'X', 'O', 'O']}
                onClick={jest.fn()}
                player= 'Rifay'
            />
        );

        const instance = component.root;
        const registraionComponent = instance.findByType(Registration);
        let tree = component.toJSON();

        expect(registraionComponent).toBeDefined();
        expect(tree).toMatchSnapshot();
    });

    test('Game Component with Board should be rendered', () => {

        useRegistrationHook.mockReturnValue({
            users:{firstPlayer: 'Yousuf', secondPlayer:'Rifay'}       
        });
        const component = renderer.create(
            <Game 
                winnerMessage="Game Draw"
                squares={['X', 'O', 'O', 'X', 'X', 'O', 'X', 'O', 'O']}
                onClick={jest.fn()}
                player= 'Rifay'
            />
          );

          const instance = component.root;
          const board = instance.findByType(Board);
          let tree = component.toJSON();

          expect(board).toBeDefined();
          expect(tree).toMatchSnapshot();
    });

});