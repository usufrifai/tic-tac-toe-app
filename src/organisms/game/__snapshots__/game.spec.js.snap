// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Game Game Component with Board should be rendered 1`] = `
<div
  className="game-container"
>
  <div>
    <h1
      style={
        Object {
          "marginTop": "30px",
          "textAlign": "center",
        }
      }
    >
       Tic Tac Toe
    </h1>
  </div>
  <div
    className="game"
  >
    <div>
      <div
        className="board-container"
      >
        <div
          className="heading"
          style={
            Object {
              "width": undefined,
            }
          }
        >
          <h3>
            Game
          </h3>
        </div>
        <div
          className="board"
        >
          <div
            className="squares"
            onClick={[Function]}
          >
            <span>
               
               
            </span>
          </div>
          <div
            className="squares"
            onClick={[Function]}
          >
            <span>
               
               
            </span>
          </div>
          <div
            className="squares"
            onClick={[Function]}
          >
            <span>
               
               
            </span>
          </div>
          <div
            className="squares"
            onClick={[Function]}
          >
            <span>
               
               
            </span>
          </div>
          <div
            className="squares"
            onClick={[Function]}
          >
            <span>
               
               
            </span>
          </div>
          <div
            className="squares"
            onClick={[Function]}
          >
            <span>
               
               
            </span>
          </div>
          <div
            className="squares"
            onClick={[Function]}
          >
            <span>
               
               
            </span>
          </div>
          <div
            className="squares"
            onClick={[Function]}
          >
            <span>
               
               
            </span>
          </div>
          <div
            className="squares"
            onClick={[Function]}
          >
            <span>
               
               
            </span>
          </div>
        </div>
        <h5
          className="playerName"
        >
          Hey Yousuf its your turn
        </h5>
      </div>
    </div>
    <div>
      <div
        className="information-container"
      >
        <div
          className="heading"
          style={
            Object {
              "width": undefined,
            }
          }
        >
          <h3>
            Information
          </h3>
        </div>
        <div
          className="list-group"
          onKeyDown={[Function]}
        >
          <div
            className="list-group-item list-group-item-warning"
            data-rb-event-key={null}
            disabled={false}
            onClick={[Function]}
          >
            Player One : 
            Yousuf
          </div>
          <div
            className="list-group-item"
            data-rb-event-key={null}
            disabled={false}
            onClick={[Function]}
          >
            Player Two : 
            Rifay
          </div>
        </div>
        <img
          alt="Winner"
          className="trophy"
          src="winner.gif"
        />
        ;
        <h4 />
        <button
          className="btn btn-warning"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          Start New Game
        </button>
         
        <button
          className="btn btn-secondary"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          Play Again
        </button>
      </div>
    </div>
  </div>
  <div
    className="history-table-container"
  >
    <div>
      <div
        className="history-table-container"
      >
        <div
          className="heading"
          style={
            Object {
              "width": "100%",
            }
          }
        >
          <h3>
            Game History
          </h3>
        </div>
        <table
          className="table table-striped table-bordered table-hover"
        >
          <thead>
            <tr>
              <th>
                #
              </th>
              <th>
                Player One
              </th>
              <th>
                Player Two
              </th>
              <th>
                Winner
              </th>
              <th>
                Date
              </th>
            </tr>
          </thead>
          <tbody />
        </table>
      </div>
    </div>
  </div>
</div>
`;

exports[`Game Game Component with Registration form should be rendered 1`] = `
<div
  className="game-container"
>
  <div>
    <h1
      style={
        Object {
          "marginTop": "30px",
          "textAlign": "center",
        }
      }
    >
       Tic Tac Toe
    </h1>
  </div>
  <div
    className="game"
  >
    <div>
      <form
        className="registration-container"
      >
        <div
          className="heading"
          style={
            Object {
              "width": undefined,
            }
          }
        >
          <h3>
            Register
          </h3>
        </div>
        <div
          className="form-group"
        >
          <label
            className="form-label"
            htmlFor="playerOne"
          >
            <h4>
               Player One 
            </h4>
          </label>
          <input
            className="field form-control"
            id="playerOne"
            onChange={[Function]}
            placeholder="First Player Name"
            required={true}
            style={
              Object {
                "fontSize": "20px",
                "height": "50px",
              }
            }
            type="text"
          />
        </div>
        <div
          className="form-group"
        >
          <label
            className="form-label"
            htmlFor="playerTwo"
            style={
              Object {
                "marginTop": "5px",
              }
            }
          >
            <h4>
               Player Two 
            </h4>
          </label>
          <input
            className="form-control"
            id="playerTwo"
            onChange={[Function]}
            placeholder="Second Player Name"
            required={true}
            style={
              Object {
                "fontSize": "20px",
                "height": "50px",
              }
            }
            type="text"
          />
        </div>
        <div
          className="form-group"
        >
          <button
            className="btn btn-warning"
            disabled={false}
            onClick={[Function]}
            style={
              Object {
                "fontSize": "20px",
                "height": "50px",
              }
            }
            type="submit"
          >
            Start Game
          </button>
        </div>
      </form>
    </div>
    <div>
      <div
        className="information-container"
      >
        <div
          className="heading"
          style={
            Object {
              "width": undefined,
            }
          }
        >
          <h3>
            Information
          </h3>
        </div>
        <div
          className="list-group"
          onKeyDown={[Function]}
        >
          <div
            className="list-group-item list-group-item-warning"
            data-rb-event-key={null}
            disabled={false}
            onClick={[Function]}
          >
            Player One : 
          </div>
          <div
            className="list-group-item"
            data-rb-event-key={null}
            disabled={false}
            onClick={[Function]}
          >
            Player Two : 
          </div>
        </div>
        <img
          alt="Winner"
          className="trophy"
          src="winner.gif"
        />
        ;
        <h4 />
        <button
          className="btn btn-warning"
          disabled={true}
          onClick={[Function]}
          type="button"
        >
          Start New Game
        </button>
         
        <button
          className="btn btn-secondary"
          disabled={true}
          onClick={[Function]}
          type="button"
        >
          Play Again
        </button>
      </div>
    </div>
  </div>
  <div
    className="history-table-container"
  >
    <div>
      <div
        className="history-table-container"
      >
        <div
          className="heading"
          style={
            Object {
              "width": "100%",
            }
          }
        >
          <h3>
            Game History
          </h3>
        </div>
        <table
          className="table table-striped table-bordered table-hover"
        >
          <thead>
            <tr>
              <th>
                #
              </th>
              <th>
                Player One
              </th>
              <th>
                Player Two
              </th>
              <th>
                Winner
              </th>
              <th>
                Date
              </th>
            </tr>
          </thead>
          <tbody />
        </table>
      </div>
    </div>
  </div>
</div>
`;
